app/models/docker_builder_service.rb

```diff
@@ -8,21 +9,19 @@ class DockerBuilderService
 
   attr_reader :build, :execution
 
-  def self.build_docker_image(dir, docker_options, output)
-    output.puts("### Creating tarfile for Docker build")
-    create_docker_tarfile(dir) do |tarfile|
-      output.puts("### Running Docker build")
-      # our image can depend on other images in the registry ... only first registry supported atm
-      credentials_to_pull = registry_credentials(DockerRegistry.first)
-
-      docker_image =
-        Docker::Image.build_from_tar(tarfile, docker_options, Docker.connection, credentials_to_pull) do |chunk|
-          output.write_docker_chunk(chunk)
-        end
-      output.puts('### Docker build complete')
+  def self.build_docker_image(dir, output)
+    TerminalExecutor.new(output).execute!(
+      "cd #{dir.shellescape}",
+      *local_docker_login,
```

Comment:
Might need to be careful about echo'ing out registry credentials, which are secrets.